File: webapp/src/types/store.ts
// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {GlobalState as ReduxGlobalState} from 'mattermost-redux/types/store';

import type combinedReducers from '../reducers';

export type GlobalState = ReduxGlobalState & {
    'plugins-jira': PluginState
};

export type PluginState = ReturnType<typeof combinedReducers>

export const pluginStateKey = 'plugins-jira' as const;

File: server/utils/kvstore/store.go
// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package kvstore

import "github.com/mattermost/mattermost-plugin-jira/server/utils/types"

type Store interface {
	KVStore

	Entity(string) EntityStore
	ValueIndex(string, types.ValueArray) ValueIndexStore
	IDIndex(string) IDIndexStore
}

type store struct {
	KVStore
}

func NewStore(kv KVStore) Store {
	return &store{
		KVStore: kv,
	}
}
